src/common/infra/config.rs

```diff
@@ -68,8 +67,6 @@ pub static REALTIME_ALERT_TRIGGERS: Lazy<RwAHashMap<String, db_scheduler::Trigge
     Lazy::new(Default::default);
 pub static ALERTS_TEMPLATES: Lazy<RwHashMap<String, Template>> = Lazy::new(Default::default);
 pub static DESTINATIONS: Lazy<RwHashMap<String, Destination>> = Lazy::new(Default::default);
-pub static DASHBOARD_REPORTS: Lazy<RwHashMap<String, reports::Report>> =
-    Lazy::new(Default::default);
```

Comment:
Do we need to be caching dashboards?

If not, we can potentially remove the cache, for now at least, and add it back if its needed later. Upon adding it back we will want to update the caching logic to use keys based on the reports' KSUIDs rather than their names.

If we need caching at the moment we can add this back as is for now. Although unique names are currently not enforced by the DB, they are enforced by the service layer's create and update logic.

Reply:
I guess, removing caching is fine for now, as far as I remember, we use this cache only for listing and get apis of reports nothing else. But check once if `DASHBOARD_REPORTS` is used anywhere else

Reply:
Sounds good. And it looks like those are the only places its used.

I'm curious, is this the kind of thing that we even want to cache?  If getting reports is a relatively infrequent operation where latency isn't super important or noticeable (eg: if the scheduler is processing certain reports on a daily/weekly basis) is it worth the memory footprint of keeping all reports in memory and the extra logic of checking/maintaining the cache?

I certainly haven't measured the memory usage myself, so it might be trivial in most cases. But as far as I can tell we don't actually have capacity limits (or eviction policies) on our in-memory caches so I suppose that there's theoretically no limit to the amount of reports we might try to hold in memory.

Reply:
Yes you are right. I do agree with you as well. I think we can go ahead with removing reports cache. It should not be used so frequently like alerts and all.